src/app/components/Header.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  AppBar,
+  Box,
+  Button,
+  IconButton,
+  SxProps,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+import React from "react";
+
+interface IHeader {
+  leftIcon?: React.ReactNode;
+  title?: string;
+  rightIcon?: React.ReactNode;
+  alignCenter?: boolean;
+}
+
+const Header = (props: IHeader) => {
+  const { title, leftIcon, rightIcon, alignCenter } = props;
+  return (
+    <AppBar position="static" elevation={0} color="inherit">
+      <Toolbar>
+        {leftIcon}
+        <Typography
+          variant="h6"
+          component="div"
+          className={`flex-grow ${alignCenter ? "text-center" : "text-start"}`}
+        >
+          {title}
+        </Typography>
+
+        {alignCenter && !rightIcon ? <Box className="w-[40px]" /> : rightIcon}
```

Comment:
a little confused about this logic, if alignCenter=false and rightIcon isn't given will rightIcon be null/undefined? i tried it and seems to be working but a little confused

Reply:
My thought process is this: 
- If alignCenter == true and there is no right icon, we need to return an empty div/box.
- if alignCenter == true and there is a right icon, we can render the right icon.
- if alignCenter == false and there is no icon, it will go to the else statement of the ternary and try to render `{rightIcon}` which would be nothing.